addons/hr_recruitment_integration_monster/wizard/hr_recruitment_post.py

```diff
@@ -0,0 +1,220 @@
+                        else 178
+                    },
+                    'Location': {
+                        'City': self.job_id.address_id.city,
+                        'State': self.job_id.address_id.state_id.name,
+                        'CountryCode': self.job_id.address_id.country_code,
+                        'PostalCode': self.job_id.address_id.zip,
+                    },
+                },
+            },
+        }
+
+        if self.api_data:
+            data['jobRefCode'] = self.api_data['jobRefCode']
+        else:
+            last_post = self.env['hr.job.post'].search(
+                domain=[
+                    ('platform', '=', self.env.ref(
+                        'hr_recruitment_integration_monster.recruitment_platform_monster'
+                    ).id),
+                    ('job_id', '=', self.job_id.id),
+                    ('api_data', '!=', False),
+                ],
+                order='id desc',
+                limit=1,
+            )
+            if last_post:
+                occurrence_str = str(int(last_post.api_data['jobRefCode'].rsplit('_', 1)[1]) + 1)
+            else:
+                occurrence_str = '1'
+
+            data['jobRefCode'] = '_'.join(
+                [str(self.job_id.id), datetime.now().strftime('%Y%m%d'), occurrence_str]
+            )
+
+        if self.job_id.monster_industry_id:
+            # not taking multiple industries into account because only one will be selected by monster
+            data['JobPostings']['JobPosting']['Industries'] = {
+                'Industry': [
+                    {
+                        'IndustryName': {
+                            'monsterId': self.job_id.monster_industry_id.monster_id
+                        },
+                    },
+                ],
+            }
+
+        if self.job_id.employment_status_id:
+            if self.job_id.employment_status_id.monster_id:
+                data['JobInformation']['JobStatus'] = {
+                    'monsterId': self.job_id.employment_status_id.monster_id,
+                }
+            if self.job_id.employment_status_id.is_flexible_hours:
+                data['JobInformation']['WorkHours'] = {'monsterId': 6}
+            elif self.job_id.employment_status_id.hours_per_week:
+                data['JobInformation']['WorkHours'] = {
+                    'monsterId': convert_hours_to_monster_id(),
+                }
+
+        if self.job_id.contract_type_id and self.job_id.contract_type_id.monster_id:
+            data['JobInformation']['JobType'] = {
+                'monsterId': self.job_id.contract_type_id.monster_id,
+            }
+
+        if self.job_id.salary_min and self.job_id.salary_max \
+            and self.job_id.salary_time_unit \
+                and self.job_id.currency_id:
+            monster_currency_unit = self.env['monster.currency.unit'].sudo().search(
+                domain=[('code', '=', self.job_id.currency_id.name)],
```

Comment:
maybe some issue with translation

Reply:
solved through the change of data models